include/boost/geometry/algorithms/detail/disjoint/interface.hpp

```diff
@@ -104,13 +102,21 @@ struct disjoint<default_strategy, false>
 } // namespace resolve_strategy
 
 
-namespace resolve_variant {
+namespace resolve_dynamic {
 
-template <typename Geometry1, typename Geometry2>
+template
+<
+    typename Geometry1, typename Geometry2,
+    bool IsDynamic = util::is_dynamic_geometry<Geometry1>::value
+                  || util::is_dynamic_geometry<Geometry2>::value,
+    bool IsGC = util::is_geometry_collection<Geometry1>::value
```

Comment:
Nitpick: since there is non-abbreviated `IsDynamic`, then `IsGC` could be `IsCollection` ;)